cmd/snap-confine/snap-confine.c

```diff
@@ -81,6 +81,9 @@ int main(int argc, char **argv)
 	if (!verify_security_tag(security_tag))
 		die("security tag %s not allowed", security_tag);
 
+	const char *snap_name = getenv("SNAP_NAME");
+	sc_snap_name_validate(snap_name, NULL);
```

Comment:
This is used (and documented) for all the functions that can return an error. It's a protocol I follow as it makes code generic (it can recover errors or can die with just one function) and easier to test (errors can be observed)